Implement the composite search bar for the mobile search header

obulat opened the pull request:
## Fixes
<!-- If PR doesn't fully resolve the issue, replace 'Fixes' below with 'Related to'. -->
<!-- If there is no issue being resolved, please consider opening one before creating this pull request. -->
Fixes #1876 by @obulat

## Description
<!-- Concisely describe what the pull request does. -->
<!-- Add screenshots, videos, or other media to show the problem and the solution when appropriate. -->
This PR implements the new VHeaderMobile component that is used on mobile widths for the search result and single result pages.
It does not use the `VSearchBar` or `VInputField` components, copying some code from them instead. Re-using the existing components is difficult because of the different border treatments (in the current component, everything is inside an outer border, and in the existing VSearchBar, some components are outside of the border), and nesting several components and several levels of template `ref`s is difficult to follow. We could extract the common functionality in the future, if that makes the code simpler.

## Testing Instructions
<!-- Give steps for the reviewer to verify that this PR fixes the problem; or delete this section entirely. -->
Go to 'http://localhost:8443/search/?q=cat&ff_new_header=on' on a narrow screen. The search bar should behave correctly. You should be able to update the search term, clear it, open (a stub of) the `VContentSettingsModal`.

You can also check out the VHeaderMobile story: https://wordpress.github.io/openverse-frontend/_preview/1875/?path=/story/components-vheader-vheadermobile-vheadermobile--default-story

## Checklist
<!-- Replace  the [ ] with [x] to check the boxes. -->
- [ ] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [ ] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [ ] My commit messages follow [best practices][best_practices].
- [ ] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [ ] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin
<!-- You must read and understand the following attestation. -->

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

zackkrida commented on src/components/VHeader/VHeaderMobile/VBackButton.vue:
```diff
@@ -0,0 +1,40 @@
+    size="disabled"
+    class="icon-button flex h-12 w-12 flex-shrink-0 rotate-90 items-center justify-center bg-tx active:shadow-ring"
+    variant="plain--avoid"
+    aria-label="Go back"
```
```suggestion
    aria-label="Go back"
```

Should this be added to `en.json`?

obulat replied:
Definitely! Thank you for catching.

zackkrida commented on src/components/VHeader/VHeaderMobile/VContentSettingsModal.vue:
```diff
@@ -0,0 +1,45 @@
+      </nav>
+    </template>
+  </VModal>
+</template>
+<script lang="ts">
```
```suggestion
</template>

<script lang="ts">
```

zackkrida commented on src/components/VHeader/VSearchBar/VClearButton.vue:
```diff
@@ -0,0 +1,25 @@
+    :icon-props="{ iconPath: closeIcon }"
+    size="smaller"
+    class="border-tx text-dark-charcoal-70"
+    :aria-label="$t('browse-page.search-form.clear')"
```
```vue
    :aria-label="$t('browse-page.search-form.clear')"
```

Curiously this doesn't appear to be working:

<img width="286" alt="CleanShot 2022-10-04 at 08 39 00@2x" src="https://user-images.githubusercontent.com/6351754/193821503-efc24c50-d5e5-42b9-b0fb-ccb31cd284a0.png">

obulat replied:
I forgot to add the string to `en.json` in this branch 😬

zackkrida commented on tsconfig.json:
```diff
@@ -78,7 +78,7 @@
     "src/components/VFilters/VLicenseExplanation.vue",
     "src/components/VHeader/VFilterButton.vue",
-    "src/components/VHeader/VSearchBar/**.vue",
+    //    "src/components/VHeader/VSearchBar/**.vue",
```
```suggestion
    //    "src/components/VHeader/VSearchBar/**.vue",
```

Comments in json 😢

obulat replied:
I didn't even realize that this is a `json` file!

zackkrida reviewed (changes requested):
Very close to lgtm. Two things I noticed (one we can wait for @panchovm) on:

- The Openverse logo and the back button have different alignments so when you click on the input the text moves to the right slightly.
- The 'clear' button appears and is active even if there is no text in the input.

zackkrida reviewed (changes requested):
- The changes to the logo/back button jumping look nice 👍 the logo is larger than advised by the mockups but when I test making it smaller locally it looks quite odd. We can review this with @panchovm later in the month.
- The 'clear' button still appears when there is no search term present. I believe this is fixed in #1880 so as long as that continues I'm okay with this
- A small thing: the back button has no interaction or focus styles. Can this be updated with the universal styles from the button audit you completed too?

obulat commented:
Changes since the last review:
- the active search bar border was replaced: now using the `ring-pink` instead of `border-pink` (so, it's now external instead of internal)
- the buttons were refactored to have a full tappable width and height of 48x48px to comply with a11y requirements. The inner colored area is still 32x32px to match the mockups. 
- Hover and focus styles added to the buttons.
- 'Clear' button is removed when there is no search term.

zackkrida reviewed (approved):
LGTM. I think I see some bugs that aren't related to the contents of this PR, which I will make separate issues for.
